### PR TITLE
permit hidden options inside groups

### PR DESCRIPTION
**Pull request by temporaer**

### CODE REVIEW

**Comment by espdev** [comment]
Thanks for contributing!

Could you add the tests for this case and fix [flake8 issues](https://travis-ci.org/click-contrib/click-option-group/jobs/653389111?utm_medium=notification&utm_source=github_status)?

**Comment by espdev** [comment]
Also, I do not quite understand how a user should interact with hidden options in some groups that have relationships among options, for example in `RequiredAllOptionGroup`. In this case we do not display the hidden options in help, but we require to specify all options in the group (and hidden options also). This may be misleading.

**Comment by temporaer** [comment]
I agree, preventing hidden options in `RequiredAllOptionGroup` makes sense. I added this and also a test. Also improved behavior when an option group has no visible options.

File: click_option_group/_core.py [hunk 1]
```diff
@@ -259,6 +268,14 @@ def handle_parse_result(self, option: GroupedOption, ctx: click.Context, opts: d
 
             raise click.UsageError(error_text, ctx=ctx)
 
+    def get_help_record(self, ctx: click.Context) -> ty.Tuple[str, str]:
+        if all(o.hidden for o in self.get_options(ctx).values()):
```
Comment (espdev) [inline click_option_group/_core.py]: Maybe we could move this check to `handle_parse_result` method?

File: click_option_group/_core.py [hunk 2]
```diff
@@ -285,6 +302,14 @@ def handle_parse_result(self, option: GroupedOption, ctx: click.Context, opts: d
 
             raise click.UsageError(error_text, ctx=ctx)
 
+    def get_help_record(self, ctx: click.Context) -> ty.Tuple[str, str]:
+        for option_name, option in self.get_options(ctx).items():
+            if option.hidden:
+                error_text = f'Hidden options ("{option_name}") are not allowed inside RequiredAllOptionGroup.'
+                raise TypeError(error_text)
+
+        return super().get_help_record(ctx)
```
Comment (espdev) [inline click_option_group/_core.py]: Maybe we could add 'hidden' attribute to `forbidden_option_attrs` list instead?

Comment (espdev) [reply click_option_group/_core.py]: And I think we can add the additional info about option name here:
https://github.com/click-contrib/click-option-group/blob/a2e9d5843a117a3607ea758a18aa48b7782681b0/click_option_group/_core.py#L29-L32

File: click_option_group/_core.py (hunk 1 above)
Comment (temporaer) [reply click_option_group/_core.py]: done

File: click_option_group/_core.py (hunk 2 above)
Comment (temporaer) [reply click_option_group/_core.py]: done. There's a slight downside that I first had to call the superclass's constructor before running the check, to get the name of the option.

**Comment by espdev** [comment]
Looks good to me. Thank you!

**Pull request merged**